File: src/main/java/com/checkers/exceptions/CannotEatException.java
package com.checkers.exceptions;

import com.checkers.board.Cell;

public class CannotEatException extends RuntimeException {

    private static final String MSG_TEMPLATE = "Can't eat from %s to %s";

    public CannotEatException(Cell sourceCell, Cell destinationCell) {
        super(MSG_TEMPLATE.formatted(sourceCell, destinationCell));
    }
}

File: src/main/kotlin/com/checkers/exception/CannotEatException.kt
package com.checkers.exception

import com.checkers.board.Cell
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(HttpStatus.BAD_REQUEST)
class CannotEatException(sourceCell: Cell, destinationCell: Cell) :
    RuntimeException("Can't eat from $sourceCell to $destinationCell")
